src/backend/cache.ts

```diff
@@ -6,14 +6,19 @@ export default class CacheStore<ValueType, KeyType extends string = string> {
   private using_in_memory: boolean
   private current_store: Store | Map<string, ValueType>
   private readonly lifespan: number | null
+  private invalidateCheck: (data: ValueType) => boolean
 
   /**
    * Creates a new cache store
    * @param filename
    * @param max_value_lifespan How long an individual entry in the store will
    *                           be cached (in minutes)
    */
-  constructor(filename: string, max_value_lifespan: number | null = 60 * 6) {
+  constructor(
+    filename: string,
+    max_value_lifespan: number | null = 60 * 6,
+    options?: { invalidateCheck?: (data: ValueType) => boolean }
```

Comment:
we should update the JSDoc above the method with this new option so it's more clear what it's used for

at least from the usage of this in this PR I'm not sure I understand why it's needed